client/my-sites/themes/index.node.js

```diff
@@ -11,21 +20,78 @@ import {
 	redirectToThemeDetails,
 } from './controller';
 import { validateFilters, validateVertical } from './validate-filters';
+import { getLanguage, getLanguageRouteParam } from 'calypso/lib/i18n-utils';
+import { setLocaleRawData } from 'calypso/state/ui/language/actions';
+
+const translationsCache = {};
+
+export function setupLocale( context, next ) {
+	if ( ! context.params.lang ) {
+		const localeDataPlaceholder = { '': {} };
+		context.store.dispatch( setLocaleRawData( localeDataPlaceholder ) ); // Reset to default locale
+
+		next();
+		return;
+	}
+	const subdomainsToLocales = Object.entries( localesToSubdomains ).reduce(
+		( acc, [ key, value ] ) => {
+			return {
+				...acc,
+				[ value ]: key,
+			};
+		},
+		{}
+	);
+	const langSlug = subdomainsToLocales[ context.params.lang ] || context.params.lang;
```

Comment:
This is odd: the `zh-cn` subdomain has mapping to two locales: `zh` and `zh-sg`. Seems like what this code tries to do is not a well-defined operation.

Reply:
This mapping was supposed to be removed, thanks for reporting it.